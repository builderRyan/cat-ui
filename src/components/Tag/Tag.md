Example

```jsx
class Example extends React.Component {
  constructor (props) {
    super(props)
    this.state = {}
  }

  render () {
    return (<div className='tag-box'>
      <Tag className='default-tag'>
        default
      </Tag>

      <Tag borderColor='#ff6c6c' bgColor='#ff6c6c' textColor='#fff'>
        default
      </Tag>

      <style>
        {`
          .tag-box {
            padding: 5px;
          }
          .default-tag {
            margin-right: 8px;
          }
        `}
      </style>
    </div>
    )
  }
}

<Example />
```
