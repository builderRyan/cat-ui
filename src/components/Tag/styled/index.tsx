import styled from 'styled-components'
import { textColor, borderColor } from '../../../styles/colors'
import { smallRadius } from '../../../styles/size'
import { TagProps } from '../Tag';

const StyledTag = styled.span`
  display: inline-block;
  border-radius: ${smallRadius};
  padding: 0 4px;
  font-size: 12px;
  color: ${(props: TagProps) => props.textColor || textColor};
  border: 1px solid ${(props: TagProps) => props.borderColor || borderColor};
  background-color: ${(props: TagProps) => props.bgColor || '#fff'}
`;

export { StyledTag }
