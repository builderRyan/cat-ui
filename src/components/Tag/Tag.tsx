import * as React from 'react'
import { Component } from 'react'
import { StyledTag } from './styled'
import * as PropTypes from 'prop-types'

export type Props = {
  textColor?: string;
  bgColor?: string;
  borderColor?: string;
}

export interface TagProps {
  /** 文字 */
  children: React.ReactNode;
  /** 文字颜色 */
  textColor?: string;
  /** 背景颜色 */
  bgColor?: string;
  /** 边框颜色 */
  borderColor?: string;
  /** 类名 */
  className?: string;
  /** 行内样式 */
  style?: React.CSSProperties;
}

function Tag({ children, ...restProps }: Partial<TagProps>) {

  return (
    <StyledTag {...restProps}>
      {children}
    </StyledTag>
  )
}

Tag.propTypes = {
  children: PropTypes.node,
  textColor: PropTypes.string,
  bgColor: PropTypes.string,
  borderColor: PropTypes.string,
  className: PropTypes.string,
  style: PropTypes.object
}


export default Tag
