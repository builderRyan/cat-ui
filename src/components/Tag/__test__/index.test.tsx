import * as React from 'react'
import * as Enzyme from 'enzyme'
import { render, mount } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import Tag from '../index'


describe('Tag', () => {
  test('renders correctly', () => {
    const wrapper = render(
      <Tag>Test</Tag>
    )
    expect(wrapper).toMatchSnapshot()
  })
  test('props:textColor', () => {
    const wrapper = render(
      <Tag textColor='#ff6c6c'>Test</Tag>
    )
    expect(wrapper).toMatchSnapshot()
  })
  test('props:bgColor', () => {
    const wrapper = render(
      <Tag bgColor='#ff6c6c'>Test</Tag>
    )
    expect(wrapper).toMatchSnapshot()
  })
  test('props:borderColor', () => {
    const wrapper = render(
      <Tag borderColor='#ff6c6c'>Test</Tag>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
