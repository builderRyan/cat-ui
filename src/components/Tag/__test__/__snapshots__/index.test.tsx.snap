// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag props:bgColor 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "sc-bdVaJa gZJQVV",
    },
    "children": Array [
      Object {
        "data": "Test",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "span",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Tag props:borderColor 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "sc-bdVaJa bVhWQC",
    },
    "children": Array [
      Object {
        "data": "Test",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "span",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Tag props:textColor 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "sc-bdVaJa ksFGyZ",
    },
    "children": Array [
      Object {
        "data": "Test",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "span",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Tag renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "sc-bdVaJa jHnbHC",
    },
    "children": Array [
      Object {
        "data": "Test",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "span",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
